docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,54 @@
+# Adding a New Metrics Provider for Dummy Endpoint
+
+To create a provider for the dummy endpoint, follow these steps:
+
+1. Fork the [Keptn repo](https://github.com/keptn/lifecycle-toolkit)
+
+2. **Define the Provider Type:** In the `metrics-operator/controllers/common/providers/common.go` file,
+ define the constant `KeptnDummyProviderType` with the value `"dummy"`.
+
+    ```go
+    const KeptnDummyProviderType = "dummy"
+    ```
+
+3. **Implement the Provider:** Create your own new folder inside the
+[metrics-operator/controllers/common/providers](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/controllers/common/providers)
+ matching the new service name: dummy and a new Go package for the dummy provider.
+  This package should contain
+ a struct that implements the `KeptnSLIProvider` interface.
+  You can follow other existing implementations,
+ such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+ as an example.
+  Ensure that you implement the `EvaluateQuery` function to fetch the metrics accurately.
+  In the implementation, make a request to the dummy endpoint and return the response.
+
+    ```go
+    {% include "./dummy-code-example.go" %}
+    ```
+
+4. **Instantiate the Provider:** In the `providers.NewProvider` function
+ in the `metrics-operator/controllers/common/providers/provider.go` file,
+ add a case for the `KeptnDummyProviderType`.
+  Instantiate the dummy provider struct and return it.
+
+    ```go
+    // Inside the providers package
+
+    // NewProvider function
+    func NewProvider(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+        switch strings.ToLower(providerType) {
+        case KeptnDummyProviderType:
+            return &dummy.KeptnDummyProvider{
+                Log:        log,
+                HttpClient: http.Client{},
+            }, nil
+        // Other cases...
+        }
+    }
+    ```
+
+5. **Add Test Cases:** Write test cases to validate your implementation and ensure it works correctly.
```

Comment:
what is the difference between point 5 and 6? I would specify here to add both unit tests and kuttl tests and maybe point to docs or code examples

Reply:
I would expect a simple code example of both kind of tests in the docs

Reply:
Alright